### ISSUE TITLE
[XF] NavigationService GoBack() Doesn't fire OnAppearing of Page

### ISSUE THREAD

**Issue by RomaRudyak** [PrismLibrary/Prism#698]
When using NavigationService for navigation and calling GoBack() OnAppearing Method of the page not invoked. In Xamarin.Forms navigation when calling PopAsync method OnAppearing invoked.

### Package info

- Platform: Xamarin.Android and Xamarin.iOS
- Prism version: Prism.Core 6.1.0, Prism.Forms 6.0.1
- Xamarin version (if applicable): Xamarin 4.1.1.3, Xamarin.Forms 2.3.0.107

### Repro steps

1. Create ContentPage
2. Override OnAppearing method
3. In OnAppearing method navigate to SomeModalPage
4. From modal page invoke GoBack()

Actual result: OnAppearing not invoked
Expected Result: OnAppearing must invoked

```csharp
public abstract class AuthContentPageBase : ContentPage
    {
        [Microsoft.Practices.Unity.Dependency]
        public INavigationService NavService { get; set; }
        [Microsoft.Practices.Unity.Dependency]
        public IAuthentication Auth { get; set; }

        protected override void OnAppearing()
        {
            base.OnAppearing();
            if (!Auth.IsLogined)
            {
                NavService.Navigate("Login", useModalNavigation: true, animated: false);
            }
        }
    }
```

**Comment by joacar** [comment]
Hi,

I know that some containers (Unity) lacks a nice way to overload and instance requested from the container and in such cases the name of the parameter must be *navigationService* (at least when constructor injection is used).

When using attributes I don't know how that will turn out but try rename *NavService* to *navigationService* and if that doesn't work try requesting it in the constructor of the view model

**Comment by RomaRudyak** [comment]
Hi joacar,

Injection thought property works good. 
Issue is in that OnAppearing not fired for page when invoked GoBack();

Probably navigation with AbsoluteUri can fix firing of  OnAppearing, but it will recreate already existing page again, and this will be overwork.

But thank you for pointing me on AbsoluteUri.

**Comment by joacar** [comment]
Hi,

I have a hunch that you don't get the _right_ instance. In the method `ViewModelLocationProvider.SetDefaultViewModelFactory(Type view, Type viewModel)` the `IPageAware.Page` property is set on the instance of `NavigationService` that is created and then this instance should be the one used. In Unity this can of overload is only possible by parameter name level hence the naming becomes crucial and should, in case of constructor parameter injection, be *navigationService*. 

So basically I think that the wrong instance of `INavigationService` is created which don't hold a reference to the page you think it does and hence the `OnAppearing` is not called.

**Comment by RomaRudyak** [comment]
Thanks a lot!!

The problem was that i register PageNavigationService in my Bootstrapper class, because I need it earlier  that it initialized, and it can be that it overrides later.
